### ISSUE TITLE
asyncio.exceptions.CancelledError

### ISSUE THREAD

**Issue by genesluna** [LucianoCirino/efficiency-nodes-comfyui#89]
After the updating to the new version I'm getting this error seconds after I start ComfyUI:

connection handler failed
Traceback (most recent call last):
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 959, in transfer_data
    message = await self.read_message()
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 1029, in read_message
    frame = await self.read_data_frame(max_size=self.max_size)
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 1104, in read_data_frame
    frame = await self.read_frame(max_size)
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 1161, in read_frame
    frame = await Frame.read(
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\framing.py", line 68, in read
    data = await reader(2)
  File "asyncio\streams.py", line 707, in readexactly
  File "asyncio\streams.py", line 501, in _wait_for_data
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\server.py", line 240, in handler
    await self.ws_handler(self)
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\server.py", line 1186, in _ws_handler
    return await cast(
  File "Z:\ComfyUI_portable_broken\ComfyUI\custom_nodes\efficiency-nodes-comfyui\tsc_utils.py", line 536, in server_logic
    async for message in websocket:
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 497, in __aiter__
    yield await self.recv()
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 568, in recv
    await self.ensure_open()
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 944, in ensure_open
    raise self.connection_closed_exc()
websockets.exceptions.ConnectionClosedError: sent 1011 (unexpected error) keepalive ping timeout; no close frame received

If I uninstall the extension the error goes away. If I roll back to the previuos release, it works fine. I already tried reinstalling comfy from zero, both the embedded and venv versions but the error persists.

When I try to process an image, I get the following error from the browser UI during execution:

Error occurred when executing KSampler (Efficient):

sent 1011 (unexpected error) keepalive ping timeout; no close frame received

  File "Z:\ComfyUI_portable_broken\ComfyUI\execution.py", line 151, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
  File "Z:\ComfyUI_portable_broken\ComfyUI\execution.py", line 81, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
  File "Z:\ComfyUI_portable_broken\ComfyUI\execution.py", line 74, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
  File "Z:\ComfyUI_portable_broken\ComfyUI\custom_nodes\efficiency-nodes-comfyui\efficiency_nodes.py", line 500, in sample
    send_command_to_frontend(startListening=False)
  File "Z:\ComfyUI_portable_broken\ComfyUI\custom_nodes\efficiency-nodes-comfyui\tsc_utils.py", line 557, in send_command_to_frontend
    asyncio.run(connected_client.send(json.dumps({
  File "asyncio\runners.py", line 44, in run
  File "asyncio\base_events.py", line 649, in run_until_complete
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 635, in send
    await self.ensure_open()
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 935, in ensure_open
    raise self.connection_closed_exc()

**Comment by LucianoCirino** [comment]
This is related to the websocket connection I am creating between my nodes and the UI failing. From where are you running ComfyUI (portable, venv, collab)?? I might have to add more logic regarding this connection so it doesn't just crap out all the nodes if an error occurs.

**Comment by LucianoCirino** [comment]
I have updated repo to handle websocket connection failures better. Please try again!
relevant commit -> https://github.com/LucianoCirino/efficiency-nodes-comfyui/commit/b1a9fb14160834aaf1642aafe66e3b3e6545ba25

**Comment by genesluna** [comment]
> This is related to the websocket connection I am creating between my nodes and the UI failing. From where are you running ComfyUI (portable, venv, collab)?? I might have to add more logic regarding this connection so it doesn't just crap out all the nodes if an error occurs.

I tried both in portable and venv. Got the same error in both.

**Comment by genesluna** [comment]
> @genesluna I have updated repo to handle websocket connection failures better. Please try again! relevant commit -> [b1a9fb1](https://github.com/LucianoCirino/efficiency-nodes-comfyui/commit/b1a9fb14160834aaf1642aafe66e3b3e6545ba25)

Thank you so much. I'll try it tomorrow in the morning.

**Comment by genesluna** [comment]
I tried the updated version and I still get the same error when starting ComfyUI:

```
connection handler failed
Traceback (most recent call last):
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 959, in transfer_data
    message = await self.read_message()
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 1029, in read_message
    frame = await self.read_data_frame(max_size=self.max_size)
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 1104, in read_data_frame
    frame = await self.read_frame(max_size)
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 1161, in read_frame
    frame = await Frame.read(
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\framing.py", line 68, in read
    data = await reader(2)
  File "asyncio\streams.py", line 707, in readexactly
  File "asyncio\streams.py", line 501, in _wait_for_data
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\server.py", line 240, in handler
    await self.ws_handler(self)
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\server.py", line 1186, in _ws_handler
    return await cast(
  File "Z:\ComfyUI_portable_broken\ComfyUI\custom_nodes\efficiency-nodes-comfyui\tsc_utils.py", line 546, in server_logic
    async for message in websocket:
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 497, in __aiter__
    yield await self.recv()
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 568, in recv
    await self.ensure_open()
  File "Z:\ComfyUI_portable_broken\python_embeded\lib\site-packages\websockets\legacy\protocol.py", line 944, in ensure_open
    raise self.connection_closed_exc()
websockets.exceptions.ConnectionClosedError: sent 1011 (unexpected error) keepalive ping timeout; no close frame received
```

The difference is I only get it once. Before it kept showing time after time. When I try to generate an image I get this message:

```
Efficiency Nodes Warning: Websocket connection failure.
Live-generated preview images from the KSampler (Efficient) may not be cleared correctly. This can lead to extra images appearing in the node's preview results when live generation preview is enabled and vae decoding is set to 'true`.
```

And then this new error in the browser:

Error occurred when executing KSampler (Efficient):

```
'NoneType' object is not subscriptable

  File "Z:\ComfyUI_portable_broken\ComfyUI\execution.py", line 151, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
  File "Z:\ComfyUI_portable_broken\ComfyUI\execution.py", line 81, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
  File "Z:\ComfyUI_portable_broken\ComfyUI\execution.py", line 74, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
  File "Z:\ComfyUI_portable_broken\ComfyUI\custom_nodes\efficiency-nodes-comfyui\efficiency_nodes.py", line 489, in sample
    preview_images = preview_images(node_images, filename_prefix)
  File "Z:\ComfyUI_portable_broken\ComfyUI\custom_nodes\efficiency-nodes-comfyui\efficiency_nodes.py", line 333, in preview_images
    filename_prefix = compute_vars(images,filename_prefix)
  File "Z:\ComfyUI_portable_broken\ComfyUI\custom_nodes\efficiency-nodes-comfyui\efficiency_nodes.py", line 324, in compute_vars
    input = input.replace("%width%", str(images[0].shape[1]))
```

**Comment by LucianoCirino** [comment]
>I tried the updated version and I still get the same error when starting ComfyUI: [...]

Attempted fixing this issue through latest commit -> https://github.com/LucianoCirino/efficiency-nodes-comfyui/commit/235e6aea015e418b890bc7cb443c5ccd4b3bb3c4


>The difference is I only get it once. Before it kept showing time after time. When I try to generate an image I get this message: [...]

This message is exactly what I WANT happen. I only want specific functionality to fail when websocket connection fails, instead of just everything just crumbling.


>And then this new error in the browser: [...]

Latest commit hopefully also fixes that -> https://github.com/LucianoCirino/efficiency-nodes-comfyui/commit/235e6aea015e418b890bc7cb443c5ccd4b3bb3c4


[Download Updated Release Here](https://github.com/LucianoCirino/efficiency-nodes-comfyui/releases/download/v1.72/efficiency-nodes-comfyuiV1-72.zip)

...

Something in your setup is causing the websocket connection to fail. That is a different topic :^)

**Comment by genesluna** [comment]
> [Download Updated Release Here](https://github.com/LucianoCirino/efficiency-nodes-comfyui/releases/download/v1.72/efficiency-nodes-comfyuiV1-72.zip)

Once more, thank you. I'll try that.

> Something in your setup is causing the websocket connection to fail. That is a different topic :^)

I'm running a clean install. The only custom node I have installed is yours. And the websocket connection only fails with this node installed. If I install the old release of efficiency-nodes, it works fine.

**Comment by LucianoCirino** [comment]
> I'm running a clean install. The only custom node I have installed is yours. And the websocket connection only fails with this node installed. If I install the old release of efficiency-nodes, it works fine.

The websocket connection was not present in older versions, in the new release the connection is **only** used is for is fixing this issue ->https://github.com/LucianoCirino/efficiency-nodes-comfyui/issues/58 (see below image as an example)

![image](https://github.com/LucianoCirino/efficiency-nodes-comfyui/assets/112517630/78e0581e-50ed-4864-abed-87717afeed5d)

Without the websocket connection being established you basically will sometimes have the extra preview image stay even after you set the 'vae_decode' setting to 'true'. It is more a minor annoyance to deal with and can actually be completely avoided by simply setting the 'preview_method' = "none".

tldr: The websocket connection not working is not a big deal and will not stop nodes from properly performing any of the important functionality
